.github/workflows/test-macos.yml

```diff
@@ -0,0 +1,34 @@
+name: Unit-tests on macOS
+
+on:
+  pull_request:
+  push:
+    branches:
+      - nightly
+      - main
+      - release/*
+  workflow_dispatch:
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+        runner:
+          - macos-12
+          - macos-m1-12
```

Comment:
Previously, we only tested M1 against Python 3.8. Should we test the whole matrix here instead?